### PR TITLE
#998 turned missing page notice into severe type instead of success

### PR DESCRIPTION
**Pull request by original-brownbear**
This solves #998 :
* Turned notice from success typo to severe type
 * Same logic used as in RsFailure, but adding the LEVEL.SEVERE to the RsFlash
 * Tightened up the exception type according to what this change allows
* Added checking the xPath, including of the class the red notice, to in the existing test for "404s"

Now looks like this:

### CODE REVIEW

**Comment by karato** [comment]
@original-brownbear Many thanks, I will find someone to review it before we merge

**Comment by karato** [comment]
@mkordas review it

**Comment by mkordas** [comment]
@original-brownbear I'm on it

File: netbout-web/src/main/java/com/netbout/rest/TkAppFallback.java [hunk 1]
```diff
@@ -74,15 +77,19 @@
...
+        throws UnsupportedEncodingException {
         return new TkFallback(
             takes,
             new FbChain(
                 new FbStatus(
                     HttpURLConnection.HTTP_NOT_FOUND,
-                    new RsFailure("page not found")
+                    new RsForward(
+                        new RsFlash("page not found", Level.SEVERE),
+                        HttpURLConnection.HTTP_MOVED_PERM
```
Comment (mkordas) [inline netbout-web/src/main/java/com/netbout/rest/TkAppFallback.java]: @original-brownbear shouldn't it be 404?

File: netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java [hunk 2]
```diff
@@ -57,6 +58,18 @@
         Manifests.read("Netbout-Documentation");
 
     /**
+     * Xpath to the 'page not found' notice.
+     */
+    private static final String MISSING =
```
Comment (mkordas) [inline netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java]: @original-brownbear can we inline this constant?

Comment (original-brownbear) [inline netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java]: @mkordas Yes, sorry I was a little unsure whether or not do inline it here. I would have inlined it, but given that you asked me to setup a constant for a long regex I assumed we had some rule about long strings here :)
Should I inline it ?

File: netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java [hunk 3]
```diff
@@ -80,7 +93,7 @@ public void downloadsStaticResources() throws Exception {
     }
 
     /**
-     * TkApp can render non-found pages.
+     * TkApp can render non-found pages while adding a severe level notice.
```
Comment (mkordas) [inline netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java]: @original-brownbear I would reword it to something like `TkApp can render non-found pages and add flash message with a severe level`

**Comment by mkordas** [comment]
@original-brownbear just two comments from my side

**Comment by original-brownbear** [comment]
@mkordas on it, sec :)

**Comment by original-brownbear** [comment]
@mkordas done, inlined constant, reworded comment. Actually turned non-found into missing to get it in one line. "Non-found" isn't really a word anyways I think :)

File: netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java (hunk 2 above)
Comment (mkordas) [inline netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java]: @original-brownbear we definitely prefer inlining here, see http://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html

Comment (mkordas) [inline netbout-web/src/test/java/com/netbout/rest/TkAppITCase.java]: @original-brownbear in another ticked I asked for constant just to be able to attach Javadoc, but this was very specific case

**Comment by mkordas** [comment]
@original-brownbear yes, it's nice now :)

**Comment by mkordas** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by mkordas** [comment]
@rultor deploy

**Comment by karato** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@mkordas 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were only two issues found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by mkordas** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@karato Quality is acceptable here.

**Comment by karato** [comment]
> @karato Quality is acceptable here.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by karato** [comment]
@mkordas just added 10 mins to @elenavolokhova (for QA), payment ID is `78558877`

thanks, paid, **25 mins** to your account, payment ID is `56cf6e613602923f6f0000bf`, task took 23 hours and 8 mins to complete

review comments (c=10) added as a bonus

+25 added to your rating, current score is: [+7220](http://www.netbout.com/b/36080?open=rating)
